diff --git a/src/pdngen/src/PdnGen.tcl b/src/pdngen/src/PdnGen.tcl
index 9780d260..f8ba8aaf 100644
--- a/src/pdngen/src/PdnGen.tcl
+++ b/src/pdngen/src/PdnGen.tcl
@@ -2062,6 +2062,22 @@ proc cut_blocked_areas {tag} {
       set stripe_locs($layer_name,$tag) [::odb::bloatSet $trimmed_set $size_by]
     }
   }
+
+  if {![dict exists $grid_data rails]} {return}
+
+  foreach layer_name [dict keys [dict get $grid_data rails]] {
+    set width [get_grid_wire_width $layer_name]
+
+    set blockages [get_blockages]
+    if {[dict exists $blockages $layer_name] && [info exists stripe_locs($layer_name,$tag)]} {
+      set stripe_locs($layer_name,$tag) [::odb::subtractSet $stripe_locs($layer_name,$tag) [dict get $blockages $layer_name]]
+
+      # Trim any shapes that are less than the width of the wire
+      set size_by [expr $width / 2 - 1]
+      set trimmed_set [::odb::shrinkSet $stripe_locs($layer_name,$tag) $size_by]
+      set stripe_locs($layer_name,$tag) [::odb::bloatSet $trimmed_set $size_by]
+    }
+  }
 }
 
 proc generate_grid_vias {tag net_name} {
